hoss/coordinate_utilities.py

```diff
@@ -0,0 +1,353 @@
+""" This module contains utility functions used for
+    coordinate variables and methods to convert the
+    coordinate variable data to x/y dimension scales
+"""
+
+from typing import Dict, Set, Tuple
+
+import numpy as np
+from netCDF4 import Dataset
+from numpy import ndarray
+from pyproj import CRS
+from varinfo import VariableFromDmr, VarInfoFromDmr
+
+from hoss.exceptions import (
+    IrregularCoordinateDatasets,
+    MissingCoordinateDataset,
+    MissingValidCoordinateDataset,
+)
+from hoss.projection_utilities import (
+    get_x_y_extents_from_geographic_points,
+)
+
+
+def get_override_projected_dimension_name(
+    varinfo: VarInfoFromDmr,
+    variable_name: str,
+) -> str:
+    """returns the x-y projection variable names that would
+    match the geo coordinate names. The `latitude` coordinate
+    variable name gets converted to 'projected_y' dimension scale
+    and the `longitude` coordinate variable name gets converted to
+    'projected_x'
+
+    """
+    override_variable = varinfo.get_variable(variable_name)
+    projected_dimension_name = ''
+    if override_variable is not None:
+        if override_variable.is_latitude():
+            projected_dimension_name = 'projected_y'
+        elif override_variable.is_longitude():
+            projected_dimension_name = 'projected_x'
+    return projected_dimension_name
+
+
+def get_override_projected_dimensions(
+    varinfo: VarInfoFromDmr,
+    variable_name: str,
+) -> list[str]:
+    """
+    Returns the projected dimensions names from coordinate variables
+    """
+    latitude_coordinates, longitude_coordinates = get_coordinate_variables(
+        varinfo, [variable_name]
+    )
+    if latitude_coordinates and longitude_coordinates:
+        # there should be only 1 lat and lon coordinate for one variable
+        override_dimensions = []
+        override_dimensions.append(
+            get_override_projected_dimension_name(varinfo, latitude_coordinates[0])
+        )
+        override_dimensions.append(
+            get_override_projected_dimension_name(varinfo, longitude_coordinates[0])
+        )
+
+    else:
+        # if the override is the variable
+        override_projected_dimension_name = get_override_projected_dimension_name(
+            varinfo, variable_name
+        )
+        override_dimensions = ['projected_y', 'projected_x']
+        if override_projected_dimension_name not in override_dimensions:
+            override_dimensions = []
+    return override_dimensions
+
+
+def get_variables_with_anonymous_dims(
+    varinfo: VarInfoFromDmr, required_variables: set[str]
+) -> bool:
+    """
+    returns the list of required variables without any
+    dimensions
+    """
```

Comment:
Two quick things here:

1) The return type in the type hint is wrong: it should be `set[str]` not `bool`.
2) It would probably be clearer in the documentation string to say "set of required variables" not "list of required variables". (I'm guessing you meant list in a generic sense, rather than the Python type, but it's a little ambiguous)